pkg/runctl/run_manager.go

```diff
@@ -126,31 +141,63 @@ func (c *runManager) prepareRunNamespace(pipelineRun k8s.PipelineRun) error {
 	return nil
 }
 
-func (c *runManager) copySecrets(targetNamespace string, secretNames []string, pipelineRun k8s.PipelineRun) error {
+func (c *runManager) copyPipelinePullSecret(pipelineRun k8s.PipelineRun) (string, error) {
+	scmCloneSecretName := pipelineRun.GetSpec().JenkinsFile.Secret
+	if scmCloneSecretName == "" {
+		return "", nil
+	}
+	random, err := utils.Random(6)
+	if err != nil {
+		return "", err
+	}
+	repoBase, err := pipelineRun.GetRepoBaseURL()
+	if err != nil {
+		return "", err
+	}
+	renameFunc := k8s.AppendNameSuffixFunc(random)
+	addTektonAnnotationFunc := k8s.SetAnnotationFunc("tekton.dev/git-0", repoBase)
+	names, err := c.copySecrets(pipelineRun.GetRunNamespace(), []string{scmCloneSecretName}, pipelineRun, false, renameFunc, stripJenkinsAnnotationsFunc, stripJenkinsLabelsFunc, addTektonAnnotationFunc)
+	if err != nil {
+		return "", err
+	}
+	return names[0], nil
+}
+
+func (c *runManager) copySecrets(targetNamespace string, secretNames []string, pipelineRun k8s.PipelineRun, dockerOnly bool, mappers ...func(*v1.Secret) *v1.Secret) ([]string, error) {
```

Comment:
Instead of a `dockerOnly` flag you may accept a `filter` func. Then define a filter func that includes Docker secret types only.

Reply:
Done